components/SignUp.vue

```diff
@@ -0,0 +1,325 @@
+<template>
+  <section :class="cls" class="yellowSection  has-text-centered">
+    <p>
+      <span>Join The Revolution</span
+      ><strong>Register To Become A Scout</strong>
+      <button
+        v-on:click="showPopup = !showPopup"
+        class="modal-button"
+        data-target="modal"
+        aria-haspopup="true"
+      >
+        Register Now
+      </button>
+    </p>
+    <div
+      id="modal"
+      v-bind:class="[showPopup ? 'modal is-active' : '', 'modal']"
+    >
+      <div class="modal-background"></div>
+      <div class="modal-content">
+        <p class="title">Join The Revolution.</p>
+        <p class="subTitle">Register To Become A Scout.</p>
+        <img src="../assets/images/popup-circle-left.png" class="popupLeft" />
+        <img src="../assets/images/popup-circle-right.png" class="popupRight" />
+        <div v-if="!sending && success">
+          <p>
+            You have Successfully Registered for Bootbag, keep an eye on your
+            emails for more details!
+          </p>
+        </div>
+        <form v-else @submit.prevent="sendEmail">
+          <input
+            v-model="email"
+            type="email"
+            placeholder="Enter email here..."
+          />
+          <span v-if="error" class="error">{{ error }}</span>
+          <button :disabled="sending" type="submit">
+            <img v-if="sending" src="~/assets/spinner.svg" />
+            <span v-else>Register Now</span>
+          </button>
+        </form>
+      </div>
+      <button
+        @click="showPopup = !showPopup"
+        class="modal-close is-large"
+        aria-label="close"
+      ></button>
+    </div>
+  </section>
+</template>
+
+<script>
+import { UserService } from '~/services'
+export default {
+  name: 'SignUp',
+  props: {
+    home: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data() {
+    return {
+      sending: false,
+      success: false,
+      email: '',
+      showPopup: false,
+      isReady: false,
+      scrolled: false,
+      error: ''
+    }
+  },
+  computed: {
+    cls() {
+      return {
+        'is-visible': this.scrolled && this.home,
+        static: !this.home
+      }
+    }
+  },
+  mounted() {
+    window.addEventListener('scroll', this.handleScroll)
+  },
+  destroyed() {
+    window.removeEventListener('scroll', this.handleScroll)
+  },
+  methods: {
+    handleScroll() {
+      this.scrolled = window.scrollY > 150
+    },
+    async sendEmail() {
+      this.sending = true
+      this.error = ''
+
+      try {
+        const res = await UserService.register(this.email)
+
+        const { status } = res
+
+        if (status === 400) {
```

Comment:
Has a check been done to see if 'email is already registered' is returned (sad path)?

Reason is because if email exists I expect to see 409 status instead

Reply:
In the function itself there is a check here - https://github.com/Sapien-Interactive/bootbag/blob/main/functions/src/api/http/public/preRegistration.ts#L35.

Reply:
Ah nice